packages/itwinui-react/src/core/Dialog/DialogMain.tsx

```diff
@@ -168,6 +168,7 @@ export const DialogMain = React.forwardRef<HTMLDivElement, DialogMainProps>(
           {
             'iui-dialog-default': styleType === 'default',
             'iui-dialog-full-page': styleType === 'fullPage',
+            'iui-dialog-visible': isOpen,
```

Comment:
Adding `*-enter-done` to classname fixed playground visibility issue but it still did not work in website. Adding this line back works for all cases.